cmd/deploy/CortexClient.go

```diff
@@ -413,6 +416,77 @@ func DeployCampaign(cortex CortexClientV6, filename string, deployable bool, ove
 	return err
 }
 
+func DeployModel(cortex CortexClientV6, filepath string) string {
+	content, err := GetJsonContent(filepath)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	model := gjson.Parse(string(content))
+	status := model.Get("status").String()
+	// models can't be created with Published status, but exported only if published. So creating without status (using default initial status) and saving again with Published status
+	if status == "Published" {
+		modelBody := model.Value().(map[string]interface{})
+		modelBody["status"] = "In development"
+		initial, _ := json.Marshal(modelBody)
+		res, err := post(&cortex, "/fabric/v4/projects/"+cortex.Project+"/models", initial)
+		if err != nil {
+			log.Println(string(res))
+			log.Fatalln(err)
+		}
+	}
+	res, err := post(&cortex, "/fabric/v4/projects/"+cortex.Project+"/models", content)
+	if err != nil {
+		log.Println(string(res))
+		log.Fatalln(err)
+	}
+	return string(res)
+}
+
+func DeployExperiment(cortex CortexClientV6, filepath string) string {
+	content, err := GetJsonContent(filepath)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	res, err := post(&cortex, "/fabric/v4/projects/"+cortex.Project+"/experiments", content)
+	if err != nil {
+		log.Println(string(res))
+		log.Fatalln(err)
+	}
+	return string(res)
+}
+
+func DeployExperimentRun(cortex CortexClientV6, filename string, repoDir string) string {
+	content, err := GetJsonContent(filename)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	run := gjson.Parse(string(content))
+	expName := run.Get("experimentName").String()
+	runId := run.Get("runId").String()
+	artifacts := run.Get("artifacts")
+	path := "/fabric/v4/projects/" + cortex.Project + "/experiments/" + url.PathEscape(expName) + "/runs"
+	// experiment run is not upsert API, so deleting and inserting
+	delete(&cortex, path+"/"+runId)
+	res, err := post(&cortex, path, content)
+	if err != nil {
+		log.Println(string(res))
+		log.Fatalln(err)
+	}
+	if artifacts.Exists() {
+		for k, v := range artifacts.Value().(map[string]interface{}) {
+			content, err := ioutil.ReadFile(filepath.Join(repoDir, ARTIFACT_DIR, v.(string)))
```

Comment:
This reads the entire file into memory before uploading? Can we use a stream instead? Might run into memory limits if the artefact is massive. Food for thought. DOn't think this needs to be done right away

Reply:
Noted. Currently its common function to do all API interaction JSON, zip or binary files, so []byte is suitable. Will refactor to use streams everywhere, or use separate method for different content type